File: Day30/Cpp/Factorial_Number_Not_Greater_Than_N.cpp
#include<bits/stdc++.h>
using namespace std;
long long findFact(long long i){
    if(i==1 || i==0) return 1;
    return i*findFact(i-1);
}

vector<long long> factorialNumbers(long long n) {
    vector<long long>v;
    for(long long i=1;i<=n;i++){
        long long ans = findFact(i);
        if(ans > n){
            return v;
        }
        v.push_back(ans);
    }
}
int main()
{
    long long n;
    cin>>n;
    vector<long long> v = factorialNumbers(n);
    for(auto i:v){
        cout<<i<<" ";
    }
    return 0;
}

File: Day30/Java/Factorial_Number_Not_Greater_Than_N.java
import java.util.ArrayList;
import java.util.List;

public class Factorial_Number_Not_Greater_Than_N {
    static Long findFact(Long i){
        if(i==0 || i==1){
            return (long)1;
        }
        return i*findFact(i-1);
    }
    public static List<Long> factorialNumbers(long n) {
        // Write Your Code Here
        List<Long> ls = new ArrayList<Long>();
        for(Long i = (long)1;i<=n;i++){
            Long ans = findFact(i);
            if(ans > n) return ls;
            ls.add(ans);
        }
        return ls;
    }
}
